plugins/uefi-capsule/fu-plugin-uefi-capsule.c

```diff
@@ -70,13 +71,30 @@ fu_plugin_get_results (FuPlugin *plugin, FuDevice *device, GError **error)
 	const gchar *tmp;
 	g_autofree gchar *err_msg = NULL;
 	g_autofree gchar *version_str = NULL;
+	g_autoptr(GError) error_local = NULL;
 
 	/* trivial case */
 	if (status == FU_UEFI_DEVICE_STATUS_SUCCESS) {
 		fu_device_set_update_state (device, FWUPD_UPDATE_STATE_SUCCESS);
 		return TRUE;
 	}
 
+	/* check if something rudely removed our BOOTXXXX entry */
+	if (!fu_uefi_bootmgr_verify_fwupd (&error_local)) {
+		/* Lenovo, grr */
+		if (fu_plugin_check_hwid (plugin, "6de5d951-d755-576b-bd09-c5cf66b27234")) {
```

Comment:
should this be applied to the device via a quirk perhaps?

Reply:
Good idea, I moved `use-legacy-bootmgr-desc` to the quirk file too.